From 846e1997b92e5999a0a6a49ddda3604737adc343 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Sun, 21 Feb 2016 14:53:50 +0800
Subject: [PATCH 1583/1691] MLK-12430 ARM: imx: enable and bypass pll1_bypass
 clk before changing arm_podf

commit 846e1997b92e5999a0a6a49ddda3604737adc343 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Before changing the ARM_PODF, the pll1_bypass clock should be enabled and
bypassed to make sure the ARM_PODF can be changed.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
(cherry picked from commit 78916c528cb424b20e87887c85246fceac81f3b4)
---
 arch/arm/mach-imx/busfreq-imx.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/busfreq-imx.c b/arch/arm/mach-imx/busfreq-imx.c
index 89a5bd6..7803ea4 100644
--- a/arch/arm/mach-imx/busfreq-imx.c
+++ b/arch/arm/mach-imx/busfreq-imx.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2011-2015 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2011-2016 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -369,6 +369,12 @@ static void enter_lpm_imx6sl(void)
 			imx_clk_set_parent(pll1_sw_clk, step_clk);
 
 			/*
+			 * Need to ensure that PLL1 is bypassed and enabled
+			 * before ARM-PODF is set.
+			 */
+			clk_set_parent(pll1_bypass_clk, pll1_bypass_src_clk);
+
+			/*
 			 * Ensure that the clock will be
 			 * at original speed.
 			 */
-- 
1.9.1

